Lesson4/Building/BuildingLibrary/Building.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BuildingLibrary {
+    
+    public class Building {
+        private int _id;
+        private int _height;
+        private int _floors;
+        private int _rooms;
+        private int _porches;
+        private static int counter = 0;
+
+        public override int GetHashCode() => id;
+
+        public int id { get => _id; }
+
+        public int height {
+            get => _height;
+            set { _height = value; }
+        }
+
+        public int floors {
+            get => _floors;
+            set { _floors = value; }
+        }
+
+        public int rooms {
+            get => _rooms;
+            set { _rooms = value; }
+        }
+
+        public int porches {
+            get => _porches;
+            set { _porches = value; }
+        }
+
+        public int FloorHeight () => (height / floors);
+
+        public int RoomsPerPorch() => (rooms / porches);
+
+        public int RoomsPerFloor() => (rooms / floors);
+
+        internal Building(int id, int height, int floors, int rooms, int porches) {
+            _id = id;
+            _height = height;
+            _floors = floors;
+            _rooms = rooms;
+            _porches = porches;
+            counter = id;
+        }
+
+        internal Building(int id) : this(id, 0, 0, 0, 0) { }
+
+        internal Building() : this(counter + 1) { }
+    }
+
+    public class Creator {
```

Comment:
```suggestion
    public static class Creator {
```
Если все методы статические, то и класс стоит сделать статическим.

Reply:
да, спасибо